src/Telomare/Eval.hs

```diff
@@ -206,15 +206,15 @@ funWrap' eval fun inp =
     Pair disp newState -> (g2s disp, Just newState)
     z                  -> ("runtime error, dumped:\n" <> show z, Nothing)
 
-funWrap :: (IExpr -> RunResult IExpr) -> IExpr -> Maybe (String, IExpr) -> IO (String, Maybe IExpr)
+funWrap :: (IExpr -> IExpr) -> IExpr -> Maybe (String, IExpr) -> IO (String, Maybe IExpr)
```

Comment:
we can actually remove this function and rename `funWrap'` to `funWrap` so IO is eliminated and a bunch of other stuff is simplified